### DISCUSSION TITLE
bankbone fix the problem?

### DISCUSSION THREAD

**Discussion by chen-del** [open-mmlab/mmsegmentation#2205] [General]
When I replaced the bankbone with movlibnetv3, the problem of understanding the port index does not correspond, can you help me and let me know how to modify it? Thanks! please
[psp_head.log](https://github.com/open-mmlab/mmsegmentation/files/9771218/psp_head.log)

**Comment by MengzhangLI** [comment]
```
  File "D:\root\betname\anconda3\envs\mmlab\lib\site-packages\mmseg\models\decode_heads\psp_head.py", line 106, in _forward_feature
    x = self._transform_inputs(inputs)
  File "D:\root\betname\anconda3\envs\mmlab\lib\site-packages\mmseg\models\decode_heads\decode_head.py", line 206, in _transform_inputs
    inputs = inputs[self.in_index]
IndexError: list index out of range
```

It usually caused by output feature maps of different backbone do not have same shape or number. You can check out your original backbone output size and shape then make sure your decoder head could match the shape and number of your new backbone output.

**Comment by chen-del** [comment]
@MengzhangLI ，Thank you, where do I need to set the output so that I can confirm the output of the bankbone feature map and print it out so that it can be matched. At the same time, if I can’t confirm it, I may have to reconfigure the step size and kernel_size of the convolution kernel, yes Well? Sorry, I haven't been in touch with this change to the backbone network for a long time, and I am not familiar with the mmse file configuration.

**Comment by MengzhangLI** [comment]
> @MengzhangLI ，Thank you, where do I need to set the output so that I can confirm the output of the bankbone feature map and print it out so that it can be matched. At the same time, if I can’t confirm it, I may have to reconfigure the step size and kernel_size of the convolution kernel, yes Well? Sorry, I haven't been in touch with this change to the backbone network for a long time, and I am not familiar with the mmse file configuration.

Sorry for late reply. I suggest you (1) download certain checkpoints we provide in model readme and (2) use IDE (like PyCharm) to debug. For example, you could add a break-point [**here**](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/models/segmentors/encoder_decoder.py#L67), and you can check the shape of `x` (i.e., the output feature map list of backbone model). The most important thing is not some detailed differences like kernal_size and step size but the **shape of feature map**. When you find the difference of shapes, then you can modify the code on your own.

Best,

**Reply by chen-del** [reply]
Okay, thank you. I'll try

**Comment by MeowZheng** [comment]
please refer https://github.com/open-mmlab/mmsegmentation/tree/master/configs/mobilenet_v3
